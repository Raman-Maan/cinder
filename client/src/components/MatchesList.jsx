import React, { Component } from 'react';
import './styles/MatchesList.css';

import MatchesListItem from '../components/MatchesListItem';

class MatchesList extends Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  render() {
    if (!this.props.matches || !this.props.matches.length) {
      return (
        <div className="MatchesList">
          <div className="empty">
            <h3 className="msg align-self-center">
              No matches yet. Get swiping!
            </h3>
          </div>
        </div>
      );
    }

    const listItems = this.props.matches.map(
      ({ id, title, subtitle, img, date }) => (
        <MatchesListItem
          key={id}
          title={title}
          subtitle={subtitle}
          img={img}
          date={date}
        />
      )
    );

    return (
      <div className="MatchesList">
        <ul>{listItems}</ul>
      </div>
    );
  }
}

export default MatchesList;
