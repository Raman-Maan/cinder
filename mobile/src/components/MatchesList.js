import React from 'react';
import { connect } from 'react-redux';
import { List, ListItem, Text, Container, Spinner } from 'native-base';
import { StyleSheet } from 'react-native';

import { matchesFetchData } from '../actions';
import MatchesListItem from './MatchesListItem';
import { loggedInUser } from '../../env';


export class MatchesList extends React.Component {
  componentDidMount() {
    this.props.fetchData(loggedInUser);
  }

  render() {
    const msg = this.props.loading
      ? 'Loading...'
      : !this.props.matches.length
        ? 'No matches yet. Get swiping!'
        : 'There was an error loading matches';

    if (
      this.props.errored ||
      (this.props.loading && !this.props.matches.length) ||
      !this.props.matches.length
    ) {
      return (
        <Container style={styles.container}>
          <Text>{msg}</Text>
        </Container>
      );
    }

    const listItems = this.props.matches.map(({
      userID, userName, userBio, primaryPic, matchDate,
    }) =>
      (<MatchesListItem
        key={userID}
        userID={userID}
        userName={userName}
        userBio={userBio}
        primaryPic={primaryPic}
        matchDate={matchDate}
      />));

    return (
      <List>
        {listItems}
      </List>
    );
  }
}

const mapStateToProps = state => ({
  matches: state.matches,
  errored: state.matchesHasErrored,
  loading: state.matchesIsLoading,
});

const mapDispatchToProps = dispatch => ({
  fetchData: userID => dispatch(matchesFetchData(userID)),
});

export default connect(mapStateToProps, mapDispatchToProps)(MatchesList);


const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
