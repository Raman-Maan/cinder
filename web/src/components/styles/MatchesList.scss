$bg-color: #ddd;
$border-color: #bbb;
$empty-msg-color: #aaa;

.MatchesList {
  width: 100%;
  height: 100%;
  max-height: 100vh;
  overflow-y: auto;

  background-color: $bg-color;
  border-right: 1px solid $border-color;

  .empty {
    display: flex;
    width: 100%;
    height: 100%;

    padding: 1em;

    justify-content: center;

    .msg {
      color: $empty-msg-color;
    }
  }

  ul {
    height: 100%;
    padding: 0;
    margin: 0;
  }
}
