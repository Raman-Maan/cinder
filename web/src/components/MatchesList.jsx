import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchAllMatches } from '../actions';
import PropTypes from 'prop-types';
import MatchesListItem from '../components/MatchesListItem';
import './styles/MatchesList.css';

export class MatchesList extends Component {
  constructor(props) {
    super(props);

    this.state = {
      active: false
    };
  }

  componentDidMount() {
    this.props.fetchData(this.props.auth.userID, this.props.auth.token);
  }

  onListItemClick(id) {
    this.setState({ active: id });
    this.props.clickHandler(id);
  }

  render() {
    const msg = this.props.loading
      ? 'Loading...'
      : !this.props.matches.length
        ? 'No matches yet. Get swiping!'
        : 'There was an error loading matches';

    if (
      this.props.errored ||
      (this.props.loading && !this.props.matches.length) ||
      !this.props.matches.length
    ) {
      return (
        <div className="MatchesList">
          <div className="empty">
            <h3 className="msg align-self-center">{msg}</h3>
          </div>
        </div>
      );
    }

    const listItems = this.props.matches.map(
      ({ userID, userName, userBio, primaryPic, matchDate }) => (
        <MatchesListItem
          key={userID}
          title={userName}
          subtitle={userBio}
          img={primaryPic}
          date={matchDate}
          active={userID === this.state.active}
          // eslint-disable-next-line react/jsx-no-bind
          onClick={this.onListItemClick.bind(this, userID)}
        />
      )
    );

    return (
      <div className="MatchesList">
        <ul>{listItems}</ul>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  matches: state.match.matches,
  errored: state.match.errored,
  loading: state.match.loading,
  auth: state.auth
});

const mapDispatchToProps = dispatch => ({
  fetchData: (userID, token) => dispatch(fetchAllMatches(userID, token))
});

MatchesList.propTypes = {
  fetchData: PropTypes.func.isRequired,
  auth: PropTypes.shape({
    isFetching: PropTypes.bool,
    isAuthenticated: PropTypes.bool.isRequired,
    token: PropTypes.string,
    userID: PropTypes.string,
    message: PropTypes.string
  }),
  clickHandler: PropTypes.func.isRequired,
  matches: PropTypes.array,
  loading: PropTypes.bool,
  errored: PropTypes.bool
};

export default connect(mapStateToProps, mapDispatchToProps)(MatchesList);
