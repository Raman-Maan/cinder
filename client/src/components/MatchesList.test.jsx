import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import MatchesList from './MatchesList';

describe('<MatchesList />', () => {
  const testMatches = ['A', 'B', 'C', 'D', 'E'].map((x, index) => ({
    id: index,
    title: x,
    subtitle: x,
    img: x,
    date: new Date(`01/0${index + 1}/2018`)
  }));

  it('renders without crashing', () => {
    // simple smoke test
    shallow(<MatchesList />);
  });

  it('renders with no matches correctly', () => {
    const component = renderer.create(<MatchesList />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('renders with matches correctly', () => {
    const component = renderer
      .create(<MatchesList matches={testMatches} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
