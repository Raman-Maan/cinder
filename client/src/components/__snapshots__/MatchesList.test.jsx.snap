// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MatchesList /> renders with matches correctly 1`] = `
<div
  className="MatchesList"
>
  <ul>
    <div
      className="MatchesListItem"
    >
      <div
        className="img-container"
      >
        <img
          alt=""
          className="circular"
          src="A"
        />
      </div>
      <div
        className="details"
      >
        <div
          className="top"
        >
          <div
            className="left"
          >
            <span
              className="title"
            >
              A
            </span>
          </div>
          <div
            className="right"
          >
            <span
              className="date"
            >
              2018-01-01
            </span>
          </div>
        </div>
        <div
          className="bottom"
        >
          <span
            className="subtitle"
          >
            A
          </span>
        </div>
      </div>
    </div>
    <div
      className="MatchesListItem"
    >
      <div
        className="img-container"
      >
        <img
          alt=""
          className="circular"
          src="B"
        />
      </div>
      <div
        className="details"
      >
        <div
          className="top"
        >
          <div
            className="left"
          >
            <span
              className="title"
            >
              B
            </span>
          </div>
          <div
            className="right"
          >
            <span
              className="date"
            >
              2018-01-02
            </span>
          </div>
        </div>
        <div
          className="bottom"
        >
          <span
            className="subtitle"
          >
            B
          </span>
        </div>
      </div>
    </div>
    <div
      className="MatchesListItem"
    >
      <div
        className="img-container"
      >
        <img
          alt=""
          className="circular"
          src="C"
        />
      </div>
      <div
        className="details"
      >
        <div
          className="top"
        >
          <div
            className="left"
          >
            <span
              className="title"
            >
              C
            </span>
          </div>
          <div
            className="right"
          >
            <span
              className="date"
            >
              2018-01-03
            </span>
          </div>
        </div>
        <div
          className="bottom"
        >
          <span
            className="subtitle"
          >
            C
          </span>
        </div>
      </div>
    </div>
    <div
      className="MatchesListItem"
    >
      <div
        className="img-container"
      >
        <img
          alt=""
          className="circular"
          src="D"
        />
      </div>
      <div
        className="details"
      >
        <div
          className="top"
        >
          <div
            className="left"
          >
            <span
              className="title"
            >
              D
            </span>
          </div>
          <div
            className="right"
          >
            <span
              className="date"
            >
              2018-01-04
            </span>
          </div>
        </div>
        <div
          className="bottom"
        >
          <span
            className="subtitle"
          >
            D
          </span>
        </div>
      </div>
    </div>
    <div
      className="MatchesListItem"
    >
      <div
        className="img-container"
      >
        <img
          alt=""
          className="circular"
          src="E"
        />
      </div>
      <div
        className="details"
      >
        <div
          className="top"
        >
          <div
            className="left"
          >
            <span
              className="title"
            >
              E
            </span>
          </div>
          <div
            className="right"
          >
            <span
              className="date"
            >
              2018-01-05
            </span>
          </div>
        </div>
        <div
          className="bottom"
        >
          <span
            className="subtitle"
          >
            E
          </span>
        </div>
      </div>
    </div>
  </ul>
</div>
`;

exports[`<MatchesList /> renders with no matches correctly 1`] = `
<div
  className="MatchesList"
>
  <div
    className="empty"
  >
    <h3
      className="msg align-self-center"
    >
      No matches yet. Get swiping!
    </h3>
  </div>
</div>
`;
